src/util/postgres_client.js

```diff
@@ -268,14 +277,60 @@ class BulkOp {
     }
 
     findAndUpdateOne(find, update) {
-        const pg_update = mongo_to_pg.convertUpdate('data', update);
-        const pg_selector = mongo_to_pg('data', find);
-        const query = `UPDATE ${this.name} SET data = ${pg_update} WHERE ${pg_selector}`;
+        let encoded_update = encode_json(this.schema, update);
+        const pg_update = mongo_to_pg.convertUpdate('data', encoded_update);
+
+        let encoded_find = encode_json(this.schema, find);
+        const pg_selector = mongo_to_pg('data', encoded_find);
+
+        let query;
+        const update_keys = encoded_update.$set && Object.keys(encoded_update.$set).filter(key => key.includes('.$.'));
+        if (update_keys && update_keys.length) {
+            query = this.findAndUpdateArray(encoded_update.$set, encoded_find);
+        } else {
+            query = `UPDATE ${this.name} SET data = ${pg_update} WHERE ${pg_selector}`;
+        }
+
         this.add_query(query);
         return this;
     }
 
+    findAndUpdateArray(update, find) {
```

Comment:
Please change the naming as some internal strig manipulation function instead of an actual interface of DB function and sort the function in a more robust way to re-use in the future.

Reply:
fixed.